### DISCUSSION TITLE
[Question] best way to connect from smart watch

### DISCUSSION THREAD

**Discussion by drunkscientist** [luc-github/ESP3D#596] [Questions]
so im writing an app for the TTGO-watch, as discussed [here](https://github.com/sharandac/My-TTGO-Watch/issues/164), and im trying to decide the best way to get data from the printer to the watch. opening a telnet connection might be fine while the user is staring at their wrist, but for basic status updates im wondering if it would be more efficient to use the webUI somehow? just have the watch pull data from a url whenever the display turns on(is there an api?)? websockets? im very new to the 'networking' part of programming so im quite unsure. my first goal is get some connection and display the printers status on the watch, eventually i would like to be able to send custom serial commands, possibly leverage the macros from the webUI

**Comment by luc-github** [comment]
in 2.1 the telnet is the way to go for remote app
in 3.0 you can also use websocket

Currently only file upload need http which is not necessary fo you if just monitoring/control

**Reply by drunkscientist** [reply]
so some of the other watch-coders have made a thing for automatically updating watch app statuses, but it only supports sending json requests, will esp3d respond to those? im sorry if its one of those "inherent to websockets" sort of things, im having a painful time learning networking code. is there a difference between connecting to the webUI vs connecting to esp3d itself?  also slightly unrelated, but there does not appear to be a good gcode for getting progress, is there yet/would you be willing to add a direct command for getting basic printers status? just something to display if printing/print time remaining

**Reply by luc-github** [reply]
esp3d is just a bridge you send commands to printer and printer answer, it is to command sender responsability to understand the printer answer
so your watch or webui are the commands sender
